File: src/main/java/au/com/anthonybruno/gymbuddy/StartServer.java
package au.com.anthonybruno.gymbuddy;

import io.javalin.Javalin;

public class StartServer {

    public static void main(String[] args) {
        Javalin app = Javalin.start(8000);
        Urls urls = new Urls(app);
        urls.setupEndpoints();
    }
}

File: backend/src/main/kotlin/dev/anthonybruno/gymbuddy/StartServer.kt
package dev.anthonybruno.gymbuddy

import dev.anthonybruno.gymbuddy.db.Database
import dev.anthonybruno.gymbuddy.util.ClassPathFile

fun loadConfig(): Config {
    return if (System.getenv("STAGE") == "production") {
        EnvPropertiesConfig()
    } else {
        FileConfig.fromClassPath(ClassPathFile("settings.properties"))
    }
}

object StartServer {

    private val CONFIG = loadConfig()

    @JvmStatic
    fun main(args: Array<String>) {
        val server = Server(Database(CONFIG.dbUsername, CONFIG.dbPassword, CONFIG.dbUrl))
        server.start(CONFIG.port)
    }
}
